### PR TITLE
Fixed commit workflow for dependabot PR helper

### PR DESCRIPTION
**Pull request by kotwanikunal**
Signed-off-by: Kunal Kotwani <kkotwani@amazon.com>

### Description
- The `activationLabel` property helps the plugin to add changes automatically to the CHANGELOG. 
- Added a new label `dependabot` to our repo, made changes to dependabot for using the new label (See: #4348) 
- The changes in this PR utilize commit options to further help with the changelog commit workflow steps to sign off the commits.

### Issues Resolved
- #4319

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/changelog_verifier.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha }}
 
       - uses: dangoslen/dependabot-changelog-helper@v1
+        with:
+          activationLabel: 'dependencies'
```
Comment (kotwanikunal) [inline .github/workflows/changelog_verifier.yml:18]: This [label is generic](https://github.com/opensearch-project/OpenSearch/pulls?q=is%3Apr+label%3Adependencies) and is used by other PRs as well. 
We could potentially introduce a new label for dependabot to make things easier.

**Review by Poojita-Raj — APPROVED** [review]

File: .github/workflows/changelog_verifier.yml (hunk 1 above)
Comment (VachaShah) [reply .github/workflows/changelog_verifier.yml:18]: @kotwanikunal You can add different labels to Dependabot PRs using https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#labels if this is going to cause conflicts with other PRs with the same labels.

Comment (kotwanikunal) [reply .github/workflows/changelog_verifier.yml]: Thanks for the suggestion. I was hoping to start a discussion. 
Created a new label `dependabot`. I will update the workflow with the same.

Comment (kotwanikunal) [reply .github/workflows/changelog_verifier.yml]: PR here: #4348

Comment (kotwanikunal) [reply .github/workflows/changelog_verifier.yml]: Removed this since the plugin defaults to `dependabot` as the activation label.

**Pull request merged**